expressions_test.go

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package substraitgo_test
+
+import (
+	"fmt"
+
+	substraitgo "github.com/substrait-io/substrait-go"
+	"github.com/substrait-io/substrait-go/proto"
+	"google.golang.org/protobuf/encoding/protojson"
+	pb "google.golang.org/protobuf/proto"
+)
+
```

Comment:
I think we need roundtrip tests (like your type tests) for various nested expressions.

Reply:
sure, i can add some more roundtrip tests for some nested expressions. I'll do that this week.

Reply:
added a roundtrip test, and a comment that I'll add more roundtrip tests after parsing is implemented since that'll make it easier to generate the nested expressions to test with.